packages/cli-tools/bin/streamr-mock-data-generate.ts

```diff
@@ -24,15 +28,29 @@ export const generate = (rate: number): void => {
             yesOrNo: Math.random() > 0.5,
             arrayOfStrings: genArray(Math.floor(Math.random() * 20), () => randomString(8)),
             arrayOfIntegers: genArray(Math.floor(Math.random() * 10), () => Math.floor(Math.random() * 100))
-
         }))
     }, rate)
 }
 
+export const generateBinary = (rate: number, minLength: number, maxLength: number): void => {
+    setInterval(() => {
+        const length = Math.floor(Math.random() * (maxLength - minLength + 1)) + minLength
+        const buffer = crypto.randomBytes(length)
+        console.info(buffer.toString('hex'))
+    }, rate)
+}
+
 createCommand()
-    .description('generate and print semi-random JSON data to stdout')
+    .description('generate and print semi-random JSON data or random binary data (hexadecimal) to stdout')
     .option('-r, --rate <n>', 'rate in milliseconds', createFnParseInt('--rate'), 500)
+    .option('--binary', 'generate binary data instead of JSON')
+    .option('--min-length <n>', 'minimum message length in bytes', createFnParseInt('--min-length'), 32)
+    .option('--max-length <n>', 'maximum message length in bytes', createFnParseInt('--max-length'), 64)
     .action((options: Options) => {
-        generate(options.rate)
+        if (options.binary) {
```

Comment:
Move `setInterval` to here so that we don't duplicate it in both `generate*()` functions

Reply:
Fixed